scripts/db/src/get-user.ts

```diff
@@ -50,48 +64,100 @@ async function getByEmail(email: string, callback: DbScriptCallback) {
+        .collection<ForumUser>('users')
+        .findOne({ 'emails.address': email })
+
+      await mongoClient.close()
+
+      if (!forumUser) {
+        return null
+      }
+
+      // Which email did we find?
+      const emailInfo = forumUser.emails.find((e) => e.address === email)
+      if (!emailInfo) {
+        // This should never happen, as they were returned by mongo because that
+        // field matched
+        throw new Error(
+          `User found by email ${email}, does not have that email (should never happen)`
+        )
+      }
 
-    /** Construct a postgres client and connect to the server */
-    const client: Client = new Client(connectionInfo)
-    await client.connect()
-
-    /** Get the person based on their email, joining on the password table */
-    const query = `
-      select
-        person.id, email, first_name, last_name, password
-      from people.person
-      join auth.password on password.person_id = person.id
-      where person.email = $1
-    `
-    const result = await client.query<PersonResult>(query, [email])
-
-    /** Close the connection */
-    await client.end()
-
-    if (result.rows.length === 0) {
-      return callback(null)
+      return {
+        id: forumUser._id,
+        username: forumUser.displayName,
+        email: emailInfo.address,
+        email_verified: emailInfo.verified,
+      }
     }
 
-    const Person = result.rows[0]
+    async function getParfitUser(): Promise<CallbackUser | null> {
+      /** Declare connection info */
+      const pgConnectionInfo: PGConnectionConfig = {
+        user: POSTGRES_USERNAME,
+        password: POSTGRES_PASSWORD,
+        host: POSTGRES_HOST,
+        database: POSTGRES_DATABASE,
+        port: POSTGRES_PORT ? parseInt(POSTGRES_PORT) : 5432,
+        ssl: TEMPLATE_DATA.pgShouldSsl,
+      }
+      /** Construct a postgres client and connect to the server */
+      const pgClient: PGClient = new PGClient(pgConnectionInfo)
+      await pgClient.connect()
 
-    /** Return the valid user back to Auth0 */
-    return callback(null, {
-      id: Person.id,
-      given_name: Person.first_name,
-      family_name: Person.last_name,
-      email: Person.email,
-    })
+      /** Get the person based on their email, joining on the password table */
+      const parfitQuery = `
+        SELECT
+          id, email, first_name, last_name
+        FROM people.person
+        where person.email = $1
+      `
+      const parfitResult = await pgClient.query<PersonResult>(parfitQuery, [
+        email,
+      ])
+
+      /** Close the connection */
+      await pgClient.end()
+
+      const Person = parfitResult.rows[0]
+
+      if (parfitResult.rows.length === 0) {
+        return null
+      }
+      return {
+        id: Person.id,
+        given_name: Person.first_name,
+        family_name: Person.last_name,
+        email: Person.email,
+      }
+    }
+
+    /** Give priority to Forum users, as the integration is newer */
```

Comment:
Does this make sense as the reason? Genuine question, it actually feels arbitrary which order they go in, and I want to understand the thinking. Intuitively it feels like the codebase with the greatest number of unmigrated users (presumably the Forum) should go first.

Reply:
The greatest number of unmigrated users who are going to be trying to log in in a given day, yeah. Which, yeah means the Forum. I can update the docstring.